lib/mobile_application_platform/security_token/service.rb

```diff
@@ -16,9 +16,19 @@ def token(application:, icn:)
         Rails.logger.info("#{config.logging_prefix} token success, application: #{application}, icn: #{icn}")
         parse_response(response, application, icn)
       rescue Common::Client::Errors::ClientError => e
+        description = case e
+                      in _msg, 500, _body
+                        'service returned Internal Server error most likely due to an invalid ICN'
```

Comment:
I wouldn't interpret errors 'most likely due to...' doesn't cut it as an error description (for example, what happens when we get this error and it's not due to an invalid ICN?). Just extract the error description and return that in an error

Reply:
Yeah I understand and I dont like the 'most likely' error description either however most of the errors coming from the API are useless and provide even less info. On this example it just return a 500 with Internal Server Error. With no other explanation. Some other errors are even worse and just return and empty body